CMakeLists.txt

```diff
@@ -1,25 +1,23 @@
 cmake_minimum_required(VERSION 3.10.2)
 
-project(a.out)
+project(a.out LANGUAGES CXX)
 
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+file(GLOB SOURCES CONFIGURE_DEPENDS "${CMAKE_SOURCE_DIR}/*.cpp")
 
-# Remove for compiler-specific features
-set(CMAKE_CXX_EXTENSIONS OFF)
+add_executable(${PROJECT_NAME} ${SOURCES})
+target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
+set_target_properties(${PROJECT_NAME} PROPERTIES CXX_EXTENSIONS OFF)
 
-string(APPEND CMAKE_CXX_FLAGS " -Wall")
-string(APPEND CMAKE_CXX_FLAGS " -Wbuiltin-macro-redefined")
-string(APPEND CMAKE_CXX_FLAGS " -pedantic")
-string(APPEND CMAKE_CXX_FLAGS " -Werror")
+target_compile_options(${PROJECT_NAME} PRIVATE  -Wall
```

Comment:
Minor nitpick, this wouldn’t work on all compilers. Additionally the compile options are usually the ones that you _need_ to compile the program. Other options can go in the presets (or at least with an if).

Reply:
Thank you @mrexodia for the review.
Well, it will work for clang and GCC, the rest does not bother me because the purpose of the repo is to show Docker-based builds. `cmake-presets` is a new feature to me, I am not sure if it suits this small repo but it was very good to learn about it!